File: Business/Models/Projects/Project.cs
namespace Business.Models.Projects;

public class Project
{
    public int Id { get; set; }
    public string ProjectName { get; set; } = null!;
    public string Description { get; set; } = null!;
    public decimal TotalCost { get; set; }
    public int CustomerId { get; set; } 
    public int StatusTypeId { get; set; } 
    public int UserId { get; set; } 
}

File: projectManagement_frontend/projectManagement/src/models/Project.ts
import { DateTime } from "ts-luxon";
import { ProjectSchedule } from "./ProjectSchedule";
import {
  emptyProjectService,
  IProjectService,
} from "../interfaces/IProjectService";

export class Project {
  constructor(
    public projectName: string,
    public description: string,
    public totalCost: number,
    public customerId: number,
    public statusTypeId: number,
    public userId: number,
    public projectSchedule: ProjectSchedule,
    public projectService: IProjectService,
    public id?: number
  ) {}
}

export const newProject = new Project(
  "",
  "",
  0,
  0,
  0,
  0,
  new ProjectSchedule(DateTime.now().toJSDate()),
  emptyProjectService
);
